### ISSUE TITLE
👋 will this package support your connectomics with 🦓CEBRA approach?

### ISSUE THREAD

**Issue by MMathisLab** [neuromodulation/py_neuromodulation#313]
Hey @timonmerk et al, firstly thanks SO MUCH for using CEBRA in your awesome paper, https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10543023/pdf/nihpp-rs3212709v1.pdf. I am currently working on teaching material, and I was wondering if you have a demo notebook, or such, for how you used CEBRA in your paper? 

_" In brief, voxel-wise correlations between decoding performance and whole-brain connectivity maps seeded from channel MNI coordinates were calculated to identify an optimal connectomic template fingerprint for movement decoding (so called connectomic decoding network map) across all subjects (Fig. 2i). This allows for an optimized a priori channel selection in realtime, by identifying the individual recording channel that has most network overlap with the optimal template. Finally, we have transformed neural features from the selected channel into a lower
dimensional embedding. For this, a five-layer convolutional neural network with a temporal filter length
of 1 s was trained using the InfoNCE (Noise-Contrastive Estimation) contrastive loss function. The
resulting embeddings showed exceptionally high consistency across subjects as investigated with linear identifiability"_

i.e., j and k here:
<img width="681" alt="Screenshot 2024-04-27 at 3 54 34 PM" src="https://github.com/neuromodulation/py_neuromodulation/assets/28102185/bc72879c-760e-4978-950a-5c04a924d9de">

Sorry if I missed it in the code, but a quick serach doesn't have `cebra` in this code base! Not a worry if  you are waiting for formal publication, etc -- totally get it -- just figured I'd ask 🥰. Thanks again for using CEBRA -- truly one of my fav. uses so far 🦓🚀

**Comment by timonmerk** [comment]
Many thanks @MMathisLab! It was really great to use CEBRA and we could also show that it improved the leave-one-patient and leave-one-cohort out performances significantly by quite a big factor over the other tested methods in the revision!

We are however still in the revision process and currently work a lot to make the repo more useable. I think we will also add an option to add CEBRA as a dev dependency. But still keep the feature-estimation only methods lightweight without having the need to install e.g. PyTorch.

But I definitely have an example notebook on my ToDo list  that shows the predictions using CEBRA without patient-individual training. Luckily the Pittsburgh cohort data was all published openly so I could add a reproducible example within that cohort. Is there a specific deadline for the teaching? Then I could prioritise this notebook and will let you know once it's upstream.

**Comment by MMathisLab** [comment]
Sorry for my delay! No major rush :) I showed them your cool preprint last week, so would be also in future years :D

**Comment by timonmerk** [comment]
Ok, finally coming back to the example @MMathisLab. I added a CEBRA [decoding example ](https://neuromodulation.github.io/py_neuromodulation/auto_examples/plot_8_cebra_example.html#sphx-glr-auto-examples-plot-8-cebra-example-py)to the documentation and examples folder where the connectomic across-patient decoding is presented. It took me quite some time since hosting pure .ipynb in sphinx-gallery isn't so straightforward as I expected!
